### PR TITLE
Feature/wsl env vars

### LINKED ISSUE #2982: AutoToolsBuildEnvironment doesn't find dependencies when using Windows Bash
Conan version: 1.4
OS: Windows 10, version 1803, build 17134.1, Windows subsystem for Linux installed (openSuse)

I tried to build the following recipe:
https://github.com/wdobbe/conan-graphicsmagick.git

On Linux the (autoconf) configure script of GraphicsMagick finds the libjpeg and libpng that Conan installed. However on Windows both libjpeg and libpng are not detected. (Note that the configure script will not complain too hard about this, and compilation will start anyway).

When I open a windows bash prompt myself, set CPPFLAGS and LDFLAGS to the locations where Conan installed libjpeg and libpng and then run the configure script, it does detect both libraries.

So it looks like the CPPFLAGS and LDFLAGS environment variables that are set up by AutoToolsBuildEnvironment are not picked up by the configure script.

### PR DESCRIPTION
**Pull request by lasote**
- [x] Refer to the issue that supports this Pull Request. Closes #2982 

- Tested manually, probably deserves a test in the integration suite. Here only a unit test.
- Missing DOCS

Read comments here: https://github.com/conan-io/conan/pull/3017

### CODE REVIEW

File: conans/client/tools/win.py [hunk 1]
```diff
@@ -427,6 +427,15 @@ def run_in_windows_bash(conanfile, bashcmd, cwd=None, subsystem=None, msys_mingw
         # This needs to be set so that msys2 bash profile will set up the environment correctly.
         env_vars = {"MSYSTEM": "MINGW32" if conanfile.settings.get_safe("arch") == "x86" else "MINGW64",
                     "MSYS2_PATH_TYPE": "inherit"}
+    elif subsystem == WSL:
+        # https://blogs.msdn.microsoft.com/commandline/2017/12/22/share-environment-vars-between-wsl-and-windows/
+        names = []
+        for var_name, value in env.items():
```
Comment (memsharded) [inline conans/client/tools/win.py]: value unused? Iterate ``for var in env``

**Comment by lasote** [comment]
Let's wait for a little for feedback.

**Comment by wdobbe** [comment]
Unfortunately, your conan branch with this commit fails to configure my [graphicsmagick](https://github.com/wdobbe/conan-graphicsmagick.git) recipe.

See my comment in issue #2982 .

File: conans/model/conan_file.py [hunk 2]
```diff
@@ -256,20 +257,25 @@ def package_info(self):
...
-                                             msys_mingw=msys_mingw)
+                                             msys_mingw=msys_mingw, env=with_env)
         if run_environment:
-            with tools.environment_append(RunEnvironment(self).vars):
-                if os_info.is_macos:
-                    command = 'DYLD_LIBRARY_PATH="%s" %s' % (os.environ.get('DYLD_LIBRARY_PATH', ''), command)
-                retcode = _run()
+            env.update(RunEnvironment(self).vars)
+            if os_info.is_macos:
+                if env.get('DYLD_LIBRARY_PATH'):
```
Comment (memsharded) [inline conans/model/conan_file.py]: 2 gets over DYLID_LIBRARY_PATH, and then using only ``[0]`` item?

File: conans/client/tools/win.py [hunk 3]
```diff
@@ -451,10 +463,10 @@ def run_in_windows_bash(conanfile, bashcmd, cwd=None, subsystem=None, msys_mingw
             # Put the build_requires and requires path at the first place inside the shell
             hack_env = ' && PATH="%s:$PATH"' % inherited_path if inherited_path else ""
 
-        for var_name, value in env.items():
-            if var_name == "PATH":
-                continue
-            hack_env += ' && %s=%s' % (var_name, value)
+            for var_name, value in env.items():
```
Comment (memsharded) [inline conans/client/tools/win.py]: This was a bug?

Comment (lasote) [reply conans/client/tools/win.py]: yes.

File: conans/model/conan_file.py (hunk 2 above)
Comment (lasote) [reply conans/model/conan_file.py]: You are right, I'm not sure what I was thinking.

File: conans/client/tools/win.py [hunk 4]
```diff
@@ -451,15 +463,14 @@ def run_in_windows_bash(conanfile, bashcmd, cwd=None, subsystem=None, msys_mingw
...
+                if var_name == "PATH":
+                    continue
+                hack_env += ' && %s=%s' % (var_name, value)
```
Comment (oliverdaniell) [inline conans/client/tools/win.py]: Are you missing quotes around value?
hack_env += ' && %s="%s"' % (var_name, value)

Comment (oliverdaniell) [reply conans/client/tools/win.py]: It's also missing an export for processes started from the shell 
```hack_env += ' && export %s="%s"' % (var_name, value)```

**Comment by SSE4** [comment]
@oliverdaniell please rebase your changes, there are conflicts

**Comment by oliverdaniell** [comment]
@SSE4 changes are from @lasote, I've just added a review

**Pull request closed without merging**